src/sentry/static/sentry/app/views/projectFilters.jsx

```diff
@@ -198,40 +200,212 @@ const LegacyBrowserFilterRow = React.createClass({
+            name="ip"
+            help={t('Separate multiple entries with a newline.')}
+            placeholder="e.g. 127.0.0.1 or 10.0.0.0/8"
+            value={this.state.formData['filters:blacklisted_ips']}
+            error={errors['filters:blacklisted_ips']}
+            onChange={this.onFieldChange.bind(this, 'filters:blacklisted_ips')} />
+        </fieldset>
+      </form>
+    );
+  }
+});
+
 const ProjectFilters = React.createClass({
   mixins: [ApiMixin],
 
   getInitialState() {
+    let until = Math.floor(new Date().getTime() / 1000);
+    let since = until - 3600 * 24 * 7;
+
     return {
+      expected: 3,
       loading: true,
       error: false,
+      statsError: false,
       filterList: [],
+      querySince: since,
+      queryUntil: until,
+      stats: null,
+      rawStatsData: null,
+      processedStats: false,
+      projectOptions: {},
+      blankStats: false,
     };
   },
 
-  componentDidMount() {
+  componentWillMount() {
     this.fetchData();
   },
 
+  componentDidUpdate(prevProps) {
+    if (!this.state.loading && !this.state.stats) {
+      this.processStatsData();
+    }
+  },
+
   fetchData() {
     let {orgId, projectId} = this.props.params;
     this.api.request(`/projects/${orgId}/${projectId}/filters/`, {
       success: (data, textStatus, jqXHR) => {
+        this.setState({filterList: data});
+      },
+      error: () => {
+        this.setState({error: true});
+      },
+      complete: () => {
+        let expected = this.state.expected - 1;
         this.setState({
-          error: false,
-          loading: false,
-          filterList: data
+          expected: expected,
+          loading: expected > 0,
         });
+      }
+    });
+
+    this.api.request(`/projects/${orgId}/${projectId}/stats/`, {
+      query: {
+        since: this.state.querySince,
+        until: this.state.queryUntil,
+        resolution: '1h',
+        stat: 'blacklisted',
+      },
+      success: (data) => {
+        this.setState({rawStatsData: data});
       },
       error: () => {
+        this.setState({error: true});
+      },
+      complete: () => {
+        let expected = this.state.expected - 1;
         this.setState({
-          error: true,
-          loading: false
+          expected: expected,
+          loading: expected > 0,
+        });
+      }
+    });
+
+    this.api.request(`/projects/${orgId}/${projectId}/`, {
+      success: (data, textStatus, jqXHR) => {
+        this.setState({projectOptions: data.options});
+      },
+      error: () => {
+        this.setState({error: true});
+      },
+      complete: () => {
+        let expected = this.state.expected - 1;
+        this.setState({
+          expected: expected,
+          loading: expected > 0,
         });
       }
     });
   },
 
+  processStatsData() {
+    let blank = true;
+    let points = this.state.rawStatsData.map(point => {
+      let [x, y] = point;
+      if (y > 0) {
+        blank = false;
+      }
+      return {
```

Comment:
It's not "super clever". It's just new syntax. If it had existed for the past 10 years, you'd be using it everywhere.